cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/OIDCUtil.java

```diff
@@ -266,14 +268,27 @@ public static RelyParty getRelyParty(String clientID) {
         return relyParties.get(clientID);
     }
     
-    public static boolean accessAllowed(RelyParty rp) {
+    public static boolean accessAllowed(RelyParty rp, Subject subject) {
         GroupURI accessGroup = rp.getAccessGroup();
         if (accessGroup == null) {
             // access group not specified, allow access
             return true;
         } else {
-            GroupClient groupClient = GroupUtil.getGroupClient(accessGroup.getServiceID());
-            return groupClient.isMember(accessGroup);
+            subject = AuthenticationUtil.augmentSubject(subject);
+            Set<GroupMemberships> groupMembershipsSet = subject.getPrivateCredentials(GroupMemberships.class);
```

Comment:
The use of private credentials in the subject to store memberships is a implementation detail/secret in the LDAP persistence layer and should not be used externally like this.  Instead, construct an LdapGroupPersistence object and call `getGroups(Role.MEMBER, GroupURI.toString())`. If the single group comes back in the collection then they are a member.  If not they are not a member.

Reply:
Reworked.